src/main/java/redis/clients/jedis/csc/CacheConnection.java

```diff
@@ -11,18 +14,35 @@
 import redis.clients.jedis.RedisProtocol;
 import redis.clients.jedis.exceptions.JedisException;
 import redis.clients.jedis.util.RedisInputStream;
+import redis.clients.jedis.util.Version;
 
 public class CacheConnection extends Connection {
 
   private final Cache cache;
   private ReentrantLock lock;
+  private static final String REDIS = "redis";
+  private static final String MIN_REDIS_VERSION = "7.4";
 
   public CacheConnection(final JedisSocketFactory socketFactory, JedisClientConfig clientConfig, Cache cache) {
     super(socketFactory, clientConfig);
 
     if (protocol != RedisProtocol.RESP3) {
       throw new JedisException("Client side caching is only supported with RESP3.");
     }
+    if (!cache.compatibilityMode()) {
+      if (server == null || version == null) {
+        Map<String, Object> helloResult = hello(encode(protocol.version()));
+        if (helloResult != null) {
+          server = (String) helloResult.get("server");
+          version = (String) helloResult.get("version");
+        }
+      }
```

Comment:
This condition is redundant. If Redis doesn't support the HELLO command, the connection will fail earlier in the execution stack and we'll never get there.